src/widgets/banner/patient-banner.component.tsx

```diff
@@ -1,25 +1,46 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 
 import dayjs from "dayjs";
-import { Button } from "carbon-components-react";
+import { Button, Tag } from "carbon-components-react";
 import { CaretDown16, CaretUp16 } from "@carbon/icons-react";
 import capitalize from "lodash-es/capitalize";
 
 import { useCurrentPatient } from "@openmrs/esm-react-utils";
+import { useTranslation } from "react-i18next";
 
 import ContactDetails from "../contact-details/contact-details.component";
 
 import placeholder from "../../assets/placeholder.png";
 import { age } from "../contact-details/age-helpers";
 import styles from "./patient-banner.scss";
+import { getVisitsForPatient } from "../visit/visit.resource";
 
 export default function PatientBanner() {
-  const [showContactDetails, setShowContactDetails] = React.useState(false);
+  const [showContactDetails, setShowContactDetails] = useState(false);
+  const [hasActiveVisit, setHasActiveVisit] = useState(false);
   const [isLoadingPatient, patient, , patientErr] = useCurrentPatient();
+  const { t } = useTranslation();
   const toggleContactDetails = () => {
     setShowContactDetails(!showContactDetails);
   };
 
+  useEffect(() => {
+    if (patient) {
+      const abortController = new AbortController();
+      const sub = getVisitsForPatient(
+        patient.id,
+        abortController,
+        "custom:(stopDatetime)"
+      ).subscribe(({ ok, data }) => {
+        if (ok) {
+          setHasActiveVisit(data.results?.some(v => v.stopDatetime == null));
```

Comment:
This warrants some discussion - it looks like if a patient has any visit that has not been ended that you are showing it as an active visit. Should we be more selective? Perhaps only if the most recent visit is active? Alternatively, it will be possible to edit/add to a retrospective visit - how will you accommodate this? In general. I think we may want to have a single point to set/test if the visit is active. @brandones what do you think?

Reply:
This is actually slightly stricter than the notion of "active visits"  in the backend (see [here](https://github.com/openmrs/openmrs-core/blob/c5001a75d55d519fb6433f512d438f3c27ca163d/api/src/main/java/org/openmrs/api/db/hibernate/HibernateVisitDAO.java#L200)), but I agree that the points here are worth discussing.

Reply:
Yeah I think if the backend has a notion of what an "active visit" is, then we should respect that.